flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/internals/KinesisDataFetcher.java

```diff
@@ -584,17 +594,34 @@ private static ShardMetricsReporter registerShardMetrics(MetricGroup metricGroup
 	//  Miscellaneous utility functions
 	// ------------------------------------------------------------------------
 
+	/**
+	 * Function to map a Kinesis shard to a Flink subtask index.
+	 */
+	public interface ShardToSubtaskIndexFn {
+		/**
+		 * Function to map a Kinesis shard to a Flink subtask index.
```

Comment:
This needs a much more stronger Javadoc explaining the contract of deterministic assignments.
See `KafkaTopicPartitionAssigner` for an example.